internal/cmp/instance_helper.go

```diff
@@ -151,6 +151,9 @@ func updateInstance(ctx context.Context, sharedClient instanceSharedClient, d *u
 
 	if d.HasChanged("snapshot") {
 		snapshot := d.GetListMap("snapshot")
+		if len(snapshot) <= 0 {
+			return fmt.Errorf("Deleting snapshot is not supported currently.")
```

Comment:
i doubt if this code will be hit as we have diff validation logic already.

Reply:
Made the changes and removed it.